File: round 1/nothingreallymatters/problem4.c
#include<stdio.h>
#include<math.h>
int main()
{
 int a,sum=0;
 scanf("%d",&a);
 while(a>9)
 {
 printf("hi"); 
  while(a!=0)
   {
    sum+=a%10;
    a=a/10;
   }
  if(sum>=10) a=sum;
 }
 if(sum!=1) printf("no");
 else printf("yes");
 return 0;
}

File: q/c/submissions/RedandBlue/problem4.java
import java.util.*;
class problem4
{
static int Digsum(int x)
{
int digit,sum=0;
while(x>0)
{
digit=x%10;
sum+=digit;
x/=10;
}
return sum;
}
public static void main(String args[])
{
Scanner sc=new Scanner(System.in);
int x=sc.nextInt();

int y=Digsum(x);
while(y>=10)
{
y=Digsum(y);
}
if(y==1)
{
System.out.print("yes");
}
else System.out.print("no");
}


}